### PR TITLE
- Adding support for GradientBoostClassification/Regression

### PR DESCRIPTION
**Pull request by harshitpandey091**
- Correcting the continuousSplit logic in DecisionTreeModelInfo.
- Taking utility/src classes from 1.6 -> 2.0 adapters
- Adding support for GradientBoostClassification/Regression

### CODE REVIEW

File: adapters-2.0/src/test/java/com/flipkart/fdp/ml/adapter/DecisionTreeClassificationModelBridgeTest.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+import com.flipkart.fdp.ml.export.ModelExporter;
+import com.flipkart.fdp.ml.importer.ModelImporter;
+import com.flipkart.fdp.ml.transformer.DecisionTreeTransformer;
+
+
+public class DecisionTreeClassificationModelBridgeTest extends SparkTestBase {
+
+
+    @Test
+    public void testDecisionTreeRegressionPrediction() {
```
Comment (akshay-sharma) [inline adapters-2.0/src/test/java/com/flipkart/fdp/ml/adapter/DecisionTreeClassificationModelBridgeTest.java]: Please add a test using this in pipeline as well. You can refer 1.6 test for the same.

File: adapters-2.0/src/test/java/com/flipkart/fdp/ml/adapter/DecisionTreeRegressionModelBridgeTest.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+import com.flipkart.fdp.ml.export.ModelExporter;
+import com.flipkart.fdp.ml.importer.ModelImporter;
+import com.flipkart.fdp.ml.transformer.DecisionTreeTransformer;
+
+/**
+ * Created by akshay.us on 8/29/16.
+ */
+public class DecisionTreeRegressionModelBridgeTest extends SparkTestBase {
+
```
Comment (akshay-sharma) [inline adapters-2.0/src/test/java/com/flipkart/fdp/ml/adapter/DecisionTreeRegressionModelBridgeTest.java]: Please add a test using this in a pipeline as well. You can refer 1.6 test for the same.

File: adapters-2.0/src/test/java/com/flipkart/fdp/ml/adapter/GradientBoostClassificationModelInfoAdapterTest.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+import com.flipkart.fdp.ml.export.ModelExporter;
+import com.flipkart.fdp.ml.importer.ModelImporter;
+import com.flipkart.fdp.ml.transformer.Transformer;
+
+/**
+ * Created by akshay.us on 8/29/16.
+ */
+public class GradientBoostClassificationModelInfoAdapterTest extends SparkTestBase {
+
```
Comment (akshay-sharma) [inline adapters-2.0/src/test/java/com/flipkart/fdp/ml/adapter/GradientBoostClassificationModelInfoAdapterTest.java]: Please add a test using this in a pipeline as well.

File: adapters-2.0/src/main/java/com/flipkart/fdp/ml/adapter/GradientBoostClassificationModelInfoAdapter.java [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class GradientBoostClassificationModelInfoAdapter extends AbstractModelInfoAdapter<GBTClassificationModel, GradientBoostModelInfo> {
+
+    private static final DecisionTreeRegressionModelInfoAdapter DECISION_TREE_ADAPTER = new DecisionTreeRegressionModelInfoAdapter();
+    
+    @Override
+    GradientBoostModelInfo getModelInfo(final GBTClassificationModel sparkRfModel) {
```
Comment (akshay-sharma) [inline adapters-2.0/src/main/java/com/flipkart/fdp/ml/adapter/GradientBoostClassificationModelInfoAdapter.java]: Please rename variable sparkRFmodel to something related to GBmodel

File: models-info/src/main/java/com/flipkart/fdp/ml/transformer/GradientBoostClassificationTransformer.java [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+		if (prediction > 0.0)
+			return 1.0;
+		else
+			return 0.0;
+    }
+
+    @Override
+    public void transform(Map<String, Object> input) {
+        double[] inp = (double[]) input.get(forest.getInputKeys().iterator().next());
+        input.put(forest.getOutputKeys().iterator().next(), predict(inp));
```
Comment (akshay-sharma) [inline models-info/src/main/java/com/flipkart/fdp/ml/transformer/GradientBoostClassificationTransformer.java]: please add raw prediction as well as class prediction as output.

**Review by akshay-sharma — CHANGES_REQUESTED** [review]
Overall, the changes look good to me. Please incorporate the said suggestions for completeness.

File: adapters-2.0/src/main/java/com/flipkart/fdp/ml/adapter/GradientBoostClassificationModelInfoAdapter.java [hunk 6]
```diff
@@ -0,0 +1,65 @@
+package com.flipkart.fdp.ml.adapter;
```
[the remark this answers is not in the document]
Comment (harshitpandey091) [reply adapters-2.0/src/main/java/com/flipkart/fdp/ml/adapter/GradientBoostClassificationModelInfoAdapter.java]: Will take this up later in another review. Will take sometime.

File: models-info/src/main/java/com/flipkart/fdp/ml/transformer/GradientBoostClassificationTransformer.java (hunk 5 above)
Comment (harshitpandey091) [reply models-info/src/main/java/com/flipkart/fdp/ml/transformer/GradientBoostClassificationTransformer.java]: GradientBoost doesn't emits rawPrediction as of now.
https://spark.apache.org/docs/latest/ml-classification-regression.html#gradient-boosted-trees-gbts

**Comment by harshitpandey091** [comment]
Added pipeline test-cases and rawPrediction checks.
Regarding GBT in 1.6, There are a few differences between 1.6 and 2.0, which will take sometime for me to completely understand. Can take that up as a separate commit in the future

**Review by akshay-sharma — APPROVED** [review]
LGTM

**Pull request merged**